types/index.d.ts

```diff
@@ -2223,6 +2223,38 @@ declare namespace math {
      */
     quantileSeq(A: MathArray | Matrix, prob: number | BigNumber | MathArray, sorted?: boolean): number | BigNumber | Unit | MathArray;
 
+    /**
+     * Compute the standard deviation of a matrix or a list with values. The
+     * standard deviations is defined as the square root of the variance:
+     * std(A) = sqrt(variance(A)). In case of a (multi dimensional) array or
+     * matrix, the standard deviation over all elements will be calculated.
+     * Optionally, the type of normalization can be specified as second
+     * parameter. The parameter normalization can be one of the following
+     * values: 'unbiased' (default) The sum of squared errors is divided by
+     * (n - 1) 'uncorrected' The sum of squared errors is divided by n
+     * 'biased' The sum of squared errors is divided by (n + 1)
+     * @param a variadic argument of number to calculate standard deviation
+     * @returns The standard deviation array
+     */
+    std(...values: number[]): number
+    /**
+     * Compute the standard deviation of a matrix or a list with values. The
+     * standard deviations is defined as the square root of the variance:
+     * std(A) = sqrt(variance(A)). In case of a (multi dimensional) array or
+     * matrix, the standard deviation over all elements will be calculated.
+     * Optionally, the type of normalization can be specified as second
+     * parameter. The parameter normalization can be one of the following
+     * values: 'unbiased' (default) The sum of squared errors is divided by
+     * (n - 1) 'uncorrected' The sum of squared errors is divided by n
+     * 'biased' The sum of squared errors is divided by (n + 1)
+     * @param array A single matrix to compute standard deviation.
+     * @param dimension A dimension to calculate standard deviation
+     * @param normalization Determines how to normalize the variance. Choose
+     * ‘unbiased’ (default), ‘uncorrected’, or ‘biased’. Default value:
+     * ‘unbiased’.
+     * @returns The standard deviation array
+     */
+    std(array: number[][] | Matrix, dimension: number, normalization?: 'unbiased' | 'uncorrected' | 'biased' | 'unbiased'): number[]
```

Comment:
I don't think 'unbiased' needs to be listed twice in the options for normalization, here and in a few places below.

Reply:
Oh, and doesn't the underlying mathjs function handle a 3-dimensional matrix (or more) and wouldn't that have input type `number[][][]` and return `number[][]`? Will such a call now be disallowed in TypeScript?

Reply:
I think `array: number[][] | Matrix` can be replaced with `array: MathArray | Matrix`, like in most functions right now.

`MathArray ` is currently defined as:

```
type MathArray = number[] | number[][]
```

In a separate topic, we could discuss what the best definition is for `MathArray` since indeed it doesn't reflect allowing n-dimensional arrays. But let's keep this PR focused on just `std` and `variance`.

Reply:
By using MathArray, it imply that code such as this is valid.
`std([1,2,3], 1)`
This is why I only allow specify dim when the data itself is a matrix.
In fact, before gwhitney replied, i didn't know that mathjs also support ndarray.
If that's the case, It'd make more sense to turn into generic. I think TensorFlow.js is a good example of this.

So back to the original intention of this pull request... Should I make it support MathArray or just number[][] or Matrix ?

Reply:
I understand your example of `std([1,2,3], 1)`, which would throw an error. But `std([1,2,3], 0)` is still valid (though it doesn't make sense in practice).

Can you please use `MathArray` for consistency? (and we can improve on `MathArray` later on)

Reply:
Done.